video/out/drm_atomic.c

```diff
@@ -231,18 +240,34 @@ struct drm_atomic_context *drm_atomic_create_context(struct mp_log *log, int fd,
         drmplane = NULL;
     }
 
-    if (!ctx->primary_plane) {
-        mp_err(log, "Failed to find primary plane\n");
-        goto fail;
+    // default OSD plane to overlay if unspecified
+    if (!ctx->osd_plane) {
```

Comment:
Does DRM guarantee that plane ID:s are never 0? If not we need to rewrite this with a separate bool flag variable.